extensions/resource-deployment/src/ui/deployClusterWizard/deployClusterWizard.ts

```diff
@@ -135,27 +144,24 @@ export class DeployClusterWizard extends WizardBase<DeployClusterWizard, DeployC
 		}
 	}
 
-	private scriptToNotebook(): void {
-		this.setEnvironmentVariables(process.env);
-		this.notebookService.launchNotebook(this.wizardInfo.notebook).then((notebook: azdata.nb.NotebookEditor) => {
-			notebook.edit((editBuilder: azdata.nb.NotebookEditorEdit) => {
-				// 5 is the position after the 'Set variables' cell in the deployment notebooks
-				editBuilder.insertCell({
-					cell_type: 'code',
-					source: this.model.getCodeCellContentForNotebook()
-				}, 5);
-			});
-		}, (error) => {
-			vscode.window.showErrorMessage(error);
-		});
+	private async scriptToNotebook(): Promise<void> {
+		this.setEnvironmentVariables(process.env, this._toolsService.toolsForCurrentProvider);
+		const variableValueStatements = this.model.getCodeCellContentForNotebook(this._toolsService.toolsForCurrentProvider);
+		const insertionPosition = 5; // Cell number 5 is the position where the python variable setting statements need to be inserted in this.wizardInfo.notebook.
+		try {
+			await this.notebookService.launchNotebookWithEdits(this.wizardInfo.notebook, variableValueStatements, insertionPosition);
+		} catch (error) {
+			vscode.window.showErrorMessage(getErrorMessage(error));
+		}
 	}
 
-	private setEnvironmentVariables(env: NodeJS.ProcessEnv): void {
+	private setEnvironmentVariables(env: NodeJS.ProcessEnv, tools: ITool[] = []): void {
 		env[VariableNames.AdminPassword_VariableName] = this.model.getStringValue(VariableNames.AdminPassword_VariableName);
 		env[VariableNames.DockerPassword_VariableName] = this.model.getStringValue(VariableNames.DockerPassword_VariableName);
 		if (this.model.authenticationMode === AuthenticationMode.ActiveDirectory) {
 			env[VariableNames.DomainServiceAccountPassword_VariableName] = this.model.getStringValue(VariableNames.DomainServiceAccountPassword_VariableName);
 		}
+		setEnvironmentVariablesForInstallPaths(tools, env);
```

Comment:
why are we passing the path using environment variables? it is only meant for passwords.

Reply:
Is there a downside to this?

Reply:
the paths are needed because tools may not have been added to the global %PATH% variable if they were installed by ads.

Reply:
Wouldn't this be a problem for all the Notebooks? I thought we told users that restarting ADS may be necessary after installing tools to pick up the updated PATH.

And couldn't you just update the PATH as well? Seems like a more standard way of handling that.

Reply:
On closer look with Alan it is not needed here at all because we are adding the necessary tools path variables directly as Python code within notebook which is avaialble both when executing from the notebook and from background execution. So this indeed superfluous here and not needed. Going to remove this.

Reply:
Removed this call.